src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+        pos_y = torch.stack((pos_y[:, :, :, 0::2].sin(), pos_y[:, :, :, 1::2].cos()), dim=4).flatten(3)
+        pos = torch.cat((pos_y, pos_x), dim=3).permute(0, 3, 1, 2)
+        return pos
+
+
+class MLP(nn.Sequential):
+    def __init__(self, input_dim: int, hidden_dim: int, output_dim: int, num_layers: int = 3):
+        """A classic Multi Layer Perceptron (MLP)
+
+        Args:
+            input_dim (int): [description]
+            hidden_dim (int): [description]
+            output_dim (int): [description]
+            num_layers (int, optional): [description]. Defaults to 3.
+        """
+        in_dims: List[int] = [input_dim] + [hidden_dim] * (num_layers - 1)
+        out_dims: List[int] = [hidden_dim] * (num_layers - 1) + [output_dim]
+
+        layers: List[nn.Module] = []
+        for i, (in_dim, out_dim) in enumerate(zip(in_dims, out_dims)):
+            # TODO should name them, e.g. fc, act ...
+            layer: nn.Module = nn.Sequential(
+                nn.Linear(in_dim, out_dim), nn.ReLU(inplace=True) if i < num_layers - 1 else nn.Identity()
+            )
+            layers.append(layer)
+
+        super().__init__(*layers)
+
+
+class MaskFormerPixelLevelModule(nn.Module):
+    def __init__(self, config: MaskFormerConfig):
+        """Pixel Level Module proposed in `Per-Pixel Classification is Not All You Need for Semantic Segmentation <https://arxiv.org/abs/2107.06278>`_. It runs the input image trough a backbone and a pixel decoder, generating a image features and pixel embeddings."""
+        super().__init__()
+        self.backbone = SwinTransformerBackbone(
+            pretrain_img_size=config.swin_pretrain_img_size,
+            patch_size=config.swin_patch_size,
+            in_chans=config.swin_in_channels,
+            embed_dim=config.swin_embed_dim,
+            depths=config.swin_depths,
+            num_heads=config.swin_num_heads,
+            window_size=config.swin_window_size,
+        )
+        self.pixel_decoder = MaskFormerPixelDecoder(
+            in_features=self.backbone.get_outputs_shape()[-1],
+            feature_size=config.fpn_feature_size,
+            mask_feature_size=config.mask_feature_size,
+            lateral_widths=self.backbone.get_outputs_shape()[:-1],
+        )
+
+    def forward(self, x: Tensor) -> Tuple[Tensor, Tensor]:
+        features: List[Tensor] = self.backbone(x)
+        # the last feature is actually the output from the last layer
+        image_features: Tensor = features[-1]
+        pixel_embeddings: Tensor = self.pixel_decoder(features)
+        return image_features, pixel_embeddings
+
+
+class MaskFormerTransformerModule(nn.Module):
+    def __init__(self, in_features: int, config: MaskFormerConfig):
+        super().__init__()
+        self.position_embedder = PositionEmbeddingSine(num_pos_feats=config.hidden_size // 2, normalize=True)
+        self.queries_embedder = nn.Embedding(config.num_queries, config.hidden_size)
```

Comment:
```suggestion
        self.position_embeddings = PositionEmbeddingSine(num_pos_feats=config.hidden_size // 2, normalize=True)
        self.queries_embeddings = nn.Embedding(config.num_queries, config.hidden_size)
```
Hmm we use the term `embeddings` for all other models currently, but to be honest embedder makes more sense :p wondering what @sgugger's opinion is here regarding naming.